packages/devextreme-react/src/core/use-option-scanning.ts

```diff
@@ -0,0 +1,159 @@
+    optionElement.descriptor.initialValuesProps,
+    optionElement.descriptor.templates,
+  );
+
+  const configCollections: Record<string, IConfigNode[]> = {};
+  const configCollectionMaps: Record<string, Record<string, number>> = {};
+  const configs: Record<string, IConfigNode> = {};
+  const templates: ITemplate[] = [];
+
+  Children.map(
+    children,
+    (child) => {
+      const elementType = getElementType(child);
+      if (elementType === ElementType.Template) {
+        const templateElement = child as ReactElement<ITemplateProps>;
+        const template = getNamedTemplate(templateElement.props);
+
+        if (template) {
+          templates.push(template);
+        }
+      }
+    },
+  );
+
+  const childComponentCounter = useRef(0);
+
+  const configNode: IConfigNode = {
+    name: optionElement.descriptor.name,
+    predefinedOptions: optionElement.descriptor.predefinedValuesProps,
+    initialOptions: separatedValues.defaults,
+    options: separatedValues.options,
+    templates,
+    configCollections,
+    configs,
+  };
+
+  const context: NestedOptionContextContent = {
+    parentExpectedChildren: optionElement.descriptor.expectedChildren,
+    parentFullName: mergeNameParts(parentFullName, optionElement.descriptor.name),
+    treeUpdateToken: updateToken,
+    getOptionComponentKey: () => {
+      childComponentCounter.current += 1;
+      return childComponentCounter.current;
+    },
+    onChildOptionsReady: (childConfigNode, childDescriptor, childUpdateToken, childComponentKey) => {
+      if (childUpdateToken !== updateToken) {
+        return;
+      }
+
+      const { isCollection, name } = childDescriptor;
+
+      if (isCollection) {
+        let collection = configCollections[name];
+        let collectionMap = configCollectionMaps[name];
+
+        if (!collection) {
+          collection = [];
+          configCollections[name] = collection;
+          collectionMap = {};
+          configCollectionMaps[name] = collectionMap;
+        }
+
+        const itemIndex = collectionMap[childComponentKey] ?? collection.length;
+        childConfigNode.index = itemIndex;
+        childConfigNode.parentNode = configNode;
+
+        if (itemIndex < collection.length) {
+          collection[itemIndex] = childConfigNode;
+        } else {
+          collectionMap[childComponentKey] = itemIndex;
+          collection.push(childConfigNode);
+        }
+
+        return;
+      }
+
+      childConfigNode.parentNode = configNode;
+      configs[name] = childConfigNode;
+    },
+  };
+
+  useLayoutEffect(() => {
+    configNode.templates = configNode.templates.filter((template) => !template.isAnonymous);
+    const hasTranscludedContent = templateContainer.childNodes.length > 0;
+
+    const getHasTranscludedContent = () => {
+      if (optionElement.descriptor.isCollection) {
+        return hasTranscludedContent;
+      }
+
+      return parentFullName.length > 0 ? hasTranscludedContent : false;
+    };
```

Comment:
- [ ] Is it really need to be a function?

```suggestion
    const hasTranscludedContent = templateContainer.childNodes.length > 0 && (
      optionElement.descriptor.isCollection || parentFullName.length > 0
    );
```